BookingsApi/BookingsApi.DAL/Commands/UpdateHearingParticipantsCommand.cs

```diff
@@ -0,0 +1,109 @@
+using BookingsApi.DAL.Commands.Core;
+using BookingsApi.DAL.Dtos;
+using BookingsApi.DAL.Exceptions;
+using BookingsApi.Domain.Participants;
+using Microsoft.EntityFrameworkCore;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace BookingsApi.DAL.Commands
+{
+    public class UpdateHearingParticipantsCommand : ICommand
+    {
+        public UpdateHearingParticipantsCommand(Guid hearingId, List<ExistingParticipantDetails> existingParticipants,
+            List<NewParticipant> newParticipants, List<Guid> removedParticipantIds, List<LinkedParticipantDto> linkedParticipants)
+        {
+            HearingId = hearingId;
+            ExistingParticipants = existingParticipants;
+            NewParticipants = newParticipants;
+            RemovedParticipantIds = removedParticipantIds;
+            LinkedParticipants = linkedParticipants ?? new List<LinkedParticipantDto>();
+        }
+
+        public List<ExistingParticipantDetails> ExistingParticipants { get; set; }
+        public List<NewParticipant> NewParticipants { get; set; }
+        public List<Guid> RemovedParticipantIds { get; set; }
+        public Guid HearingId { get; set; }
+        public List<LinkedParticipantDto> LinkedParticipants { get; set; }
+    }
+    
+    public class UpdateHearingParticipantsCommandHandler : ICommandHandler<UpdateHearingParticipantsCommand>
+    {
+        private readonly BookingsDbContext _context;
+        private readonly IHearingService _hearingService;
+
+        public UpdateHearingParticipantsCommandHandler(BookingsDbContext context, IHearingService hearingService)
+        {
+            _context = context;
+            _hearingService = hearingService;
+        }
+
+        public async Task Handle(UpdateHearingParticipantsCommand command)
+        {
+            var hearing = await _context.VideoHearings
+                .Include(x => x.Participants).ThenInclude(x=> x.Person.Organisation)
+                .Include(x => x.Participants).ThenInclude(x => x.HearingRole.UserRole)
+                .Include(x => x.Participants).ThenInclude(x => x.CaseRole)
+                .Include(x => x.Participants).ThenInclude(x => x.LinkedParticipants)
+                .Include(h => h.Endpoints).ThenInclude(x => x.DefenceAdvocate)
+                .SingleOrDefaultAsync(x => x.Id == command.HearingId);
+            
+            if (hearing == null)
+            {
+                throw new HearingNotFoundException(command.HearingId);
+            }
+
+            _context.Update(hearing);
+            
+            await _hearingService.AddParticipantToService(hearing, command.NewParticipants);
+
+            var participants = hearing.GetParticipants().ToList();
+
+            foreach (var newExistingParticipantDetails in command.ExistingParticipants)
```

Comment:
The new at the start of newExistingParticipantDetails should be removed right?

Reply:
It should be new, as in the existing participant details are new. I did originally think of naming it updatedExistingParticipantDetails but thought that would confuse with an object that has made a database call and been updated.

Am happy to change it to that if you think it is less confusing than this.

Reply:
Ah I get it now. Yeah it's a bit of a tough one actually. I think it's probably best as it is